client/app/services/query.js

```diff
@@ -293,6 +294,21 @@ function QueryResource($resource, $http, $q, $location, currentUser, QueryResult
     return this.getParameters().get();
   };
 
+  Query.prototype.getLocalParametersDefs = function getLocalParametersDefs() {
+    if (!this.$localParameters) {
+      this.$localParameters = this.getParametersDefs().filter(p => !p.global);
```

Comment:
It must be cached or angular can't detect that the state isn't changing. The normal parameters are cached, so why is this case any different?

https://github.com/getredash/redash/pull/1504/files/cb2e6fddf44d32fc22d54e5f2a2adffcbb44dce1#diff-35fc04aab5b4f115145224036f087c3cR285